src/connection-manager/connections.ts

```diff
@@ -243,8 +245,14 @@ export default async function startConnections(): Promise<void> {
     await fetchAmendmentInfo()
     await clearConnectionsDb()
     await createConnections()
+    await backtrackAmendmentStatus()
     setInterval(() => {
+      void fetchAmendmentInfo()
       void createConnections()
     }, CM_INTERVAL)
+
+    setInterval(() => {
+      void backtrackAmendmentStatus()
```

Comment:
Technically speaking you need to have the ability to cancel a previous request in the event this request isn’t finished by the next interval tick (even if unlikely, it could happen)